src/nvim/option.c

```diff
@@ -2030,7 +2030,13 @@ void set_init_1(void)
     } else {
       free(p_enc);
       p_enc = save_enc;
+      /* mb_init() failed. Call it again for the saved encoding */
+      mb_init();
```

Comment:
I feel much better about what's going on here now. I updated https://github.com/neovim/neovim/pull/2106#issuecomment-77659442 with a description of what I understand to be the root problem.

Your fix is correct :+1: Thanks a lot!